### PR TITLE
feat: consul registry support acl(#1168)

### LINKED ISSUE #1168: 使用consul作为注册中心，不支持配置acl
### Your question

使用consul作为注册中心时，不支持配置acl，sofaboot中也没有配置项支持

坐标: com.alipay.sofa.rpc.registry.consul.ConsulRegistry

建议增加

### Environment

- SOFARPC version: 5.7.7
- JVM version (e.g. `java -version`): jdk8
- OS version (e.g. `uname -a`): linux
- Maven version: 3.8
- IDE version: idea

### PR DESCRIPTION
**Pull request by Kunple-w**
### Motivation:

consul registry support acl 

### Modification:

1. consul registry support parameter `token`;
2. test unit for consul registry acl;

### Result:

Fixes #1168

### CODE REVIEW

File: registry/registry-consul/src/test/java/com/alipay/sofa/rpc/registry/consul/ConsulRegistryAclTest.java [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+            Assert.assertEquals(0, healthServices.getValue().size());
+        }, 10, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testRegisterWithoutToken() {
+        ProviderConfig<?> providerConfig = providerConfig("consul-test-1", 12200, 12201, 12202);
+        registry.register(providerConfig);
+
+        ConsulClient consulClient = new ConsulClient("localhost", consul.getHttpPort());
```
Comment (EvenLjj) [inline registry/registry-consul/src/test/java/com/alipay/sofa/rpc/registry/consul/ConsulRegistryAclTest.java:121]: There are some places in this piece that are repeated with `testRegister`.Is there any other intention to explain, if not, can delete it or delete `testRegister`. The following withoutToken tests have the same problem.

Comment (Kunple-w) [reply registry/registry-consul/src/test/java/com/alipay/sofa/rpc/registry/consul/ConsulRegistryAclTest.java:121]: I want to make sure that my commit still works on miss token, but it looks like  `ConsulRegistryTest` job,not `ConsulRegistryAclTest`.
I will delete some code  like `testRegister`.

**Review by EvenLjj — APPROVED** [review]
LGTM

**Review by JervyShi — APPROVED** [review]
LGTM

**Pull request merged**